tests/storage/clone.go

```diff
@@ -213,9 +297,45 @@ var _ = SIGDescribe("[Serial]VirtualMachineClone Tests", func() {
 				Expect(vmClone.Status.RestoreName).To(BeNil())
 			})
 
+			It("simple clone with snapshot source", func() {
+				By("Creating a VM")
+				sourceVM = createVM()
+				Eventually(func() virtv1.VirtualMachinePrintableStatus {
+					sourceVM, err = virtClient.VirtualMachine(sourceVM.Namespace).Get(sourceVM.Name, &v1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					return sourceVM.Status.PrintableStatus
+				}, 30*time.Second, 1*time.Second).Should(Equal(virtv1.VirtualMachineStatusStopped))
+
+				By("Creating a snapshot from VM")
+				snapshot := createSnapshot(sourceVM)
+				snapshot = waitSnapshotReady(snapshot)
```

Comment:
dont know if its worth another test case but supposedly you allow clone to start even if snapshot is not ready yet, so its possible to either remove the wait for snapshot to be ready to see it works anyways in this case cant delete the vm yet, and it wont be deterministic (you wont know if the clone started before or after snapshot completed), or you can make another test where you add a dummy webhook to keep snapshot in progress example in test: 6952 then start the clone and then delete the webhook.
or just not test it, cause again not sure if its such a must :)

Reply:
That's a great observation!
Because of what's explained in [the comment above](https://github.com/kubevirt/kubevirt/pull/8458#discussion_r985511529), I can't use the function from snapshot tests to deny the volume snapshot. But see the `TODO` in code that we should use it once it's refactored.

In addition, I've replaced the call for `waitSnapshotReady` with `waitSnapshotContentsExist` since this is a requirement from the validation webhook.